### PR TITLE
`file/*` - porting over to hashicorp/go-azure-sdk's base layer 

### PR DESCRIPTION
**Pull request by catriona-m**

### CODE REVIEW

File: storage/2020-08-04/file/files/api.go [hunk 1]
```diff
@@ -4,25 +4,22 @@ import (
...
-	GetResourceID(accountName, shareName, directoryName, filePath string) string
-	ListRanges(ctx context.Context, accountName, shareName, path, fileName string) (result ListRangesResult, err error)
-	GetProperties(ctx context.Context, accountName, shareName, path, fileName string) (result GetResult, err error)
-	Delete(ctx context.Context, accountName, shareName, path, fileName string) (result autorest.Response, err error)
-	Create(ctx context.Context, accountName, shareName, path, fileName string, input CreateInput) (result autorest.Response, err error)
-	CopyAndWait(ctx context.Context, accountName, shareName, path, fileName string, input CopyInput, pollDuration time.Duration) (result CopyResult, err error)
+	PutByteRange(ctx context.Context, shareName string, path string, fileName string, input PutByteRangeInput) (resp PutRangeResponse, err error)
+	GetByteRange(ctx context.Context, shareName string, path string, fileName string, input GetByteRangeInput) (resp GetByteRangeResponse, err error)
+	ClearByteRange(ctx context.Context, shareName string, path string, fileName string, input ClearByteRangeInput) (resp ClearByteRangeResponse, err error)
```
Comment (tombuildsstuff) [inline storage/2020-08-04/file/files/api.go:12]: (I realise the original code did this, but) we don't need to name each of the outputs fwiw:

```suggestion
	PutByteRange(ctx context.Context, shareName string, path string, fileName string, input PutByteRangeInput) (PutRangeResponse, error)
	GetByteRange(ctx context.Context, shareName string, path string, fileName string, input GetByteRangeInput) (GetByteRangeResponse, error)
	ClearByteRange(ctx context.Context, shareName string, path string, fileName string, input)
```

(etc)

File: storage/2020-08-04/file/shares/api.go [hunk 2]
```diff
@@ -2,23 +2,20 @@ package shares
...
-	GetSnapshot(ctx context.Context, accountName, shareName, snapshotShare string) (result GetSnapshotPropertiesResult, err error)
-	GetStats(ctx context.Context, accountName, shareName string) (result GetStatsResult, err error)
-	GetACL(ctx context.Context, accountName, shareName string) (result GetACLResult, err error)
-	SetMetaData(ctx context.Context, accountName, shareName string, metaData map[string]string) (result autorest.Response, err error)
-	GetMetaData(ctx context.Context, accountName, shareName string) (result GetMetaDataResult, err error)
-	SetProperties(ctx context.Context, accountName, shareName string, properties ShareProperties) (result autorest.Response, err error)
-	DeleteSnapshot(ctx context.Context, accountName, shareName string, shareSnapshot string) (result autorest.Response, err error)
-	CreateSnapshot(ctx context.Context, accountName, shareName string, input CreateSnapshotInput) (result CreateSnapshotResult, err error)
-	GetResourceID(accountName, shareName string) string
+	SetACL(ctx context.Context, shareName string, input SetAclInput) (resp setAclResponse, err error)
```
Comment (tombuildsstuff) [inline storage/2020-08-04/file/shares/api.go:8]: (as above) we don't need to name the return args here:

```suggestion
	SetACL(ctx context.Context, shareName string, input SetAclInput) (setAclResponse, error)
```

also `setAclResponse` should be public

**Review by tombuildsstuff — APPROVED** [review]
Left a few comments inline but otherwise 👍

**Comment by catriona-m** [comment]
> Left a few comments inline but otherwise 👍

Thanks @tombuildsstuff, I think I've fixed these now with the exception of the lro poller which we will do in another pr.

**Review by stephybun — APPROVED** [review]
Thanks @catriona-m LGTM 👍

File: storage/2020-08-04/file/files/range_get_file.go [hunk 3]
```diff
@@ -8,35 +8,45 @@ import (
...
 	if err != nil {
-		result = file.Response
+		resp.HttpResponse = file.HttpResponse
 		err = e
 		return
 	}
 
 	if file.ContentLength == nil {
-		err = fmt.Errorf("Content-Length was nil!")
+		err = fmt.Errorf("Content-Length was nil")
```
Comment (tombuildsstuff) [inline storage/2020-08-04/file/files/range_get_file.go:36]: we probably also want to set the http response here:

```suggestion
		resp.HttpResponse = file.HttpResponse
		err = fmt.Errorf("Content-Length was nil")
```

**Review by tombuildsstuff — APPROVED** [review]
hey @catriona-m 

Apologies for the delayed re-review here - I've left some comments inline, but this otherwise LGTM 👍

File: storage/2020-08-04/file/directories/metadata_set.go [hunk 4]
```diff
@@ -6,97 +6,85 @@ import (
...
+	opts := client.RequestOptions{
+		ContentType: "application/xml; charset=utf-8",
+		ExpectedStatusCodes: []int{
+			http.StatusOK,
+		},
+		HttpMethod: http.MethodPut,
+		OptionsObject: SetMetaDataOptions{
+			metaData: input.MetaData,
+		},
+		Path: fmt.Sprintf("%s/%s", shareName, path),
```
[the remark this answers is not in the document]
Comment (catriona-m) [reply storage/2020-08-04/file/directories/metadata_set.go:50]: it seems to work as expected with or without them, but for consistency we can add it to all of them

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Pull request merged**